man/foreign.doc

```diff
@@ -1917,19 +1917,26 @@ argument is bound, the code below may fail before reaching the end of
 the word list, but even if the unification succeeds, this code avoids a
 duplicate (garbage) list and a deep unification.
 
+Note that PL_unify_list() is not used with \arg{env} but with
+\exam{env_ref}, which has a reference to \arg{env}. The only thing
+that is allowed to be done with and argument to a foreign predicate
+(such as \arg{env}) is unification; for anything that might over-write
+the term, you must use a reference created by PL_copy_term_ref().
+
 \begin{code}
 foreign_t
 pl_get_environ(term_t env)
-{ term_t item = PL_new_term_ref();
+{ term_t env_ref = PL_copy_term_ref(env);
```

Comment:
I consider _ref_ a but dubious.   It creates a new term reference that is really a copy.   They both point at the same term.    I often use the word _tail_ for the copy when it is used to iterate over a list these days.

Reply:
pl__write_atoms uses "list". I'll make them both the same: "tail".
(It's hard to think of a good name for this; and my brain for some reason refuses to form an intuitive picture of what's going on, so I have to work it out each time.)